UnitySDK/Assets/ML-Agents/Examples/3DBall/Scripts/Ball3DAgent.cs

```diff
@@ -63,7 +68,20 @@ public override void AgentReset()
         ballRb.velocity = new Vector3(0f, 0f, 0f);
         ball.transform.position = new Vector3(Random.Range(-1.5f, 1.5f), 4f, Random.Range(-1.5f, 1.5f))
                                       + gameObject.transform.position;
+        //Reset the parameters when the Agent is reset.
+        SetResetParameters();
```

Comment:
I don't think this is worth calling into CollectObservations. I think calling it in AgentReset should be enough.

Reply:
I don't think i am calling this in CollectObservations. That call to SetResetParameters is in AgentReset.